code-challenges.test.js

```diff
@@ -35,14 +90,91 @@ const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+describe("onlyOddSort", () => {
+    it("returns a sorted array of only odd numbers based on input array", () => {
+        expect(onlyOddSort(fullArr1)).toEqual([-9, 7, 9, 199])
+        expect(onlyOddSort(fullArr2)).toEqual([-823, 7, 23])
+    })
+})
+
+// ---Initial Jest Test Result---
+
+// FAIL  ./code-challenges.test.js
+// onlyOddSort
+//   ✕ returns a sorted array of only odd numbers based on input array
+
+// ● onlyOddSort › returns a sorted array of only odd numbers based on input array
+
+//   ReferenceError: onlyOddSort is not defined
+
+//     85 | describe("onlyOddSort", () => {
+//     86 |     it("returns a sorted array of only odd numbers based on input array", () => {
+//   > 87 |         expect(onlyOddSort(fullArr1)).toEqual([-9, 7, 9, 199])
+//        |         ^
+//     88 |         expect(onlyOddSort(fullArr2)).toEqual([-823, 7, 23])
+//     89 |     })
+//     90 | })
+
+//     at Object.<anonymous> (code-challenges.test.js:87:9)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
+// Time:        0.143 s, estimated 1 s
 
 // b) Create the function that makes the test pass.
 
+// Create a function called onlyOddSort with a parameter called array
+// Create variable which is equal to the argument array filtered for typeof
+// Create another variable equal to that filtered array which filters the array further into only odd numbers
+// Return that array sorted so that the numbers will appear low to high
+// Call the function
+const onlyOddSort = (array) => {
+    arr1 = array.filter(value => typeof value === 'number')
+    arr2 = arr1.filter(value => value % 2 !== 0)
+    return arr2.sort((a, b) => a - b)
+}
+
+onlyOddSort(fullArr1)
+onlyOddSort(fullArr2)
```

Comment:
Since your test is where the function is being invoked, there is no need for you to invoke the function elsewhere.